### PR TITLE
rpc: clamp zero batch concurrency

### PR DESCRIPTION
**Pull request by HoustonOla35**
## Summary

- Clamp RPC batch concurrency to at least one worker when constructing the server.
- Prevent `rpc.batch.concurrency=0` from creating an unbuffered admission channel that blocks batch processing before any request goroutine can start.
- Add a direct server constructor regression test for the zero-value case.

### CODE REVIEW

File: rpc/server.go [hunk 1]
```diff
@@ -65,6 +67,9 @@ type Server struct {
 
 // NewServer creates a new server instance with no registered handlers.
 func NewServer(batchConcurrency uint, traceRequests, debugSingleRequest, disableStreaming bool, logger log.Logger, rpcSlowLogThreshold time.Duration) *Server {
+	if batchConcurrency == 0 {
```
Comment (AskAlexSharov) [inline rpc/server.go:70]: Probably it's breaking change: 
- in the past `0` did mean `unlimited`
- now it means `1`

**Review by AskAlexSharov — CHANGES_REQUESTED** [review]

File: rpc/server.go (hunk 1 above)
Comment (AskAlexSharov) [reply rpc/server.go:70]: or `0` did mean `0` and nothing worked?

**Review by AskAlexSharov — APPROVED** [review]

**Comment by AskAlexSharov** [comment]
Add to cli flag description: `0 means 1`
